runtime/src/main/scala/akka/grpc/GrpcClientSettings.scala

```diff
@@ -21,50 +21,82 @@ import scala.concurrent.duration.{ Duration, _ }
+   * Create a client with the given service discovery mechanism. Default configuration
+   * is loaded from reference.conf
+   *
+   * @param serviceName               Name to look up in serviceDiscovery
+   * @param defaultPort               Port to use if service discovery only returns a host name
+   * @param serviceDiscoveryMechanism Service discovery mechanism to use. Must be correctly configured in the ActorSystem's config
    */
-  def apply(serviceName: String, defaultPort: Int, serviceDiscovery: SimpleServiceDiscovery, resolveTimeout: FiniteDuration) =
-    new GrpcClientSettings(serviceName, serviceDiscovery, defaultPort, resolveTimeout)
+  def apply(serviceName: String, defaultPort: Int, serviceDiscoveryMechanism: String)(implicit actorSystem: ActorSystem): GrpcClientSettings = {
+    val defaultServiceConfig = actorSystem.settings.config.getConfig("akka.grpc.client").getConfig("\"*\"")
+      .withValue("service-name", ConfigValueFactory.fromAnyRef(serviceName))
+      .withValue("port", ConfigValueFactory.fromAnyRef(defaultPort))
+      .withValue("service-discovery-mechanism", ConfigValueFactory.fromAnyRef(serviceDiscoveryMechanism))
+    GrpcClientSettings(defaultServiceConfig)
+  }
 
   /**
    * Scala API
    *
-   * @param serviceName of the service to lookup config from the ActorSystem's config
+   * Look up client settings from an ActorSystem's configuration. Client configuration
+   * must be under `akka.grpc.client`. Each client configuration falls back to the
+   * defaults defined in reference.conf
+   *
+   * @param clientName of the client configuration to lookup config from the ActorSystem's config
    */
-  def apply(serviceName: String, sys: ActorSystem): GrpcClientSettings = {
-    val akkaGrpcClientConfig = sys.settings.config.getConfig("akka.grpc.client")
-    val serviceConfig = {
+  def apply(clientName: String)(implicit actorSystem: ActorSystem): GrpcClientSettings = {
+    val akkaGrpcClientConfig = actorSystem.settings.config.getConfig("akka.grpc.client")
+    val clientConfig = {
       // Use config named "*" by default
       val defaultServiceConfig = akkaGrpcClientConfig.getConfig("\"*\"")
-      // Override "*" config with specific service config if available
-      if (akkaGrpcClientConfig.hasPath('"' + serviceName + '"'))
-        akkaGrpcClientConfig.getConfig('"' + serviceName + '"').withFallback(defaultServiceConfig)
-      else
-        defaultServiceConfig
+      require(akkaGrpcClientConfig.hasPath('"' + clientName + '"'), s"Config path `akka.grpc.client.$clientName` does not exist")
+      akkaGrpcClientConfig.getConfig('"' + clientName + '"').withFallback(defaultServiceConfig)
     }
 
-    GrpcClientSettings(serviceConfig)
+    GrpcClientSettings(clientConfig)
   }
 
-  /** Scala API */
-  def apply(config: Config): GrpcClientSettings = {
-    GrpcClientSettings(config getString "host", config getInt "port")
+  /**
+   * Scala API
```

Comment:
Please avoid single line "... API" lines as first line since this shows up as "highlighted" line in scaladoc, so its not very useful; `Scala API: Configure client via the provided Config. See reference.conf for configuration properties.` is nicer